lib/SetImageChannelMetadataPass.cpp

```diff
@@ -57,14 +51,15 @@ void concatWithFunctionMetadata(Function *F, MetadataVector &MDs) {
   }
 }
 
-void setMetadata(Module &M, ImageGetterMap &map, OffsetMap &off_map, int pc) {
+unsigned setMetadata(Module &M, Function *F, ImageGetterMap &map) {
   auto i32 = IntegerType::get(M.getContext(), 32);
 
-  for (auto order : map) {
-    auto F = order.first.first;
-    auto image = order.first.second;
+  unsigned int count = 0;
+  for (auto elem : map) {
```

Comment:
This is still non-deterministic. Pointers are effectively random values. Hash maps are always non-deterministic no matter the key, but ordered maps are non-deterministic for pointer keys.